packages/cron/src/jobs/pins-backup.js

```diff
@@ -0,0 +1,248 @@
+        /** @type {import('./bindings').RemoteBackup} */
+        const backup = { ...bak, backupUrl }
+        yield backup
+      }
+    }
+  }
+
+  /**
+   * @param {import('@aws-sdk/client-s3').S3Client} s3
+   * @param {string} bucketName
+   * @param {import('./bindings').BackupContent} bak
+   */
+  async s3Upload (s3, bucketName, bak) {
+    const key = `complete/${bak.contentCid}/${bak.sourceCid}.car`
+    const region = await s3.config.region()
+    const url = new URL(`https://${bucketName}.s3.${region}.amazonaws.com/${key}`)
+    this.log(`uploading to ${url}`)
+
+    try {
+      // Request the head object of the file we are about to backup
+      // If it throws a NotFound error then we know we need to upload it
+      await s3.headObject({ Bucket: bucketName, Key: key }).promise()
+    } catch (err) {
+      if (err.name === 'NotFound') {
+        const upload = new Upload({
+          client: s3,
+          params: {
+            Bucket: bucketName,
+            Key: key,
+            Body: Readable.from(bak.content),
+            Metadata: { structure: 'Complete' }
+          }
+        })
+        await upload.done()
+      }
+    }
+    this.log('done')
+    return url
+  }
+
+  /**
+   * @param {() => Promise<import('ipfs-core').IPFS>} getIpfs
+   * @param {Object} [options]
+   * @param {number} [options.maxDagSize] Skip DAGs that are bigger than this.
+   */
+  exportCar (ipfs, options = {}) {
+    /**
+     * @param {AsyncIterable<import('./bindings').BackupCandidate>} source
+     * @returns {AsyncIterableIterator<import('./bindings').BackupContent>}
+     */
+    return async function * (source) {
+      for await (const candidate of source) {
+        yield { ...candidate, content: this.ipfsDagExport(ipfs, candidate.sourceCid, options) }
+      }
+    }
+  }
+
+  /**
+   * Export a CAR for the passed CID.
+   *
+   * @param {import('./ipfs-client').IpfsClient} ipfs
+   * @param {import('multiformats').CID} cid
+   * @param {Object} [options]
+   * @param {number} [options.maxDagSize]
+   */
+  async * ipfsDagExport (ipfs, cid, options) {
+    const maxDagSize = options.maxDagSize || this.MAX_DAG_SIZE
+
+    let reportInterval
+    try {
+      this.log('determining size...')
+      let bytesReceived = 0
+      const bytesTotal = await this.getSize(ipfs, cid)
+      this.log(bytesTotal == null ? 'unknown size' : `size: ${this.fmt(bytesTotal)} bytes`)
+
+      if (bytesTotal != null && bytesTotal > maxDagSize) {
+        throw Object.assign(
+          new Error(`DAG too big: ${this.fmt(bytesTotal)} > ${this.fmt(maxDagSize)}`),
+          { code: 'ERR_TOO_BIG' }
+        )
+      }
+
+      reportInterval = setInterval(() => {
+        const formattedTotal = bytesTotal ? this.fmt(bytesTotal) : 'unknown'
+        this.log(`received ${this.fmt(bytesReceived)} of ${formattedTotal} bytes`)
+      }, this.REPORT_INTERVAL)
+
+      for await (const chunk of ipfs.dagExport(cid, { timeout: this.BLOCK_TIMEOUT })) {
```

Comment:
`dagExport` is not implemented on the IPFS Cluster client https://github.com/nftstorage/ipfs-cluster#api